pkg/server/sandbox_run.go

```diff
@@ -452,13 +453,40 @@ func (c *criService) generateSandboxContainerSpec(id string, config *runtime.Pod
 	}
 	g.SetProcessOOMScoreAdj(adj)
 
+	for passthroughAnnotationKey, passthroughAnnotationvalue := range c.getPassthroughAnnotations(config.Annotations, runtimePodAnnotations) {
+		g.AddAnnotation(passthroughAnnotationKey, passthroughAnnotationvalue)
+	}
+
 	g.AddAnnotation(annotations.ContainerType, annotations.ContainerTypeSandbox)
 	g.AddAnnotation(annotations.SandboxID, id)
 	g.AddAnnotation(annotations.SandboxLogDir, config.GetLogDirectory())
 
 	return g.Config, nil
 }
 
+func (c *criService) getPassthroughAnnotations(sandboxAnnotations map[string]string,
+	runtimePodAnnotations []string) (passthroughAnnotations map[string]string) {
+	passthroughAnnotations = make(map[string]string)
+
+	for podAnnoationKey, podAnnoationValue := range sandboxAnnotations {
+		for _, r := range runtimePodAnnotations {
+			if r == podAnnoationKey {
+				passthroughAnnotations[podAnnoationKey] = podAnnoationValue
+				continue
+			}
+			if strings.HasSuffix(r, "*") {
```

Comment:
Why not just do regexp? Doing this by ourselves just makes this more error prone.

Reply:
Done

Reply:
Sorry, I thought more about this, and found that regular expression actually doesn't serve the domain name matching purpose very well, because we use `.` in domain name a lot, and in regular expression if we want to match `.`, we have to use `\.`.

I think your original solution is better. @harche /cc @mikebrow WDYT?
Sorry for the churns.

Reply:
^com\.github\.containers\.virtcontainers\.